src/interface.jl

```diff
@@ -0,0 +1,252 @@
+using Distributions, Bijectors
+using ForwardDiff
+using Tracker
+
+import Base: inv, ∘
+
+import Random: AbstractRNG
+import Distributions: logpdf, rand, rand!, _rand!, _logpdf
+
+#######################################
+# AD stuff "extracted" from Turing.jl #
+#######################################
+
+abstract type ADBackend end
+struct ForwardDiffAD <: ADBackend end
+struct TrackerAD <: ADBackend end
+
+const ADBACKEND = Ref(:forward)
+function setadbackend(backend_sym)
+    @assert backend_sym == :forward_diff || backend_sym == :reverse_diff
+    backend_sym == :forward_diff && CHUNKSIZE[] == 0 && setchunksize(40)
+    ADBACKEND[] = backend_sym
+end
+
+ADBackend() = ADBackend(ADBACKEND[])
+ADBackend(T::Symbol) = ADBackend(Val(T))
+function ADBackend(::Val{T}) where {T}
+    if T === :forward_diff
+        return ForwardDiffAD
+    else
+        return TrackerAD
+    end
+end
+
+######################
+# Bijector interface #
+######################
+
+abstract type Bijector end
+abstract type ADBijector{AD} <: Bijector end
+
+struct Inversed{B <: Bijector} <: Bijector
+    orig::B
+end
+
+Broadcast.broadcastable(b::Bijector) = Ref(b)
+
+"Computes the log(abs(det(J(x)))) where J is the jacobian of the transform."
+logabsdetjac(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} = 
+    error("`logabsdetjac(b::$T1, y::$T2)` is not implemented.")
+
+"Transforms the input using the bijector."
+transform(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} =
+    error("`transform(b::$T1, y::$T2)` is not implemented.")
+
+"Computes both `transform` and `logabsdetjac` in one forward pass."
+forward(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} = 
+    error("`forward(b::$T1, y::$T2)` is not implemented.")
+
+
+transform(b::Bijector) = x -> transform(b, x)
+(ib::Inversed{<: Bijector})(y) = transform(ib, y)
+
+# default `forward` implementations; should in general implement efficient way
+# of computing both `transform` and `logabsdetjac` together.
+forward(b::Bijector, x) = (rv=transform(b, x), logabsdetjac=logabsdetjac(b, x))
+forward(ib::Inversed{<: Bijector}, y) = (rv=transform(ib, y), logabsdetjac=logabsdetjac(ib, y))
+
+# defaults implementation for inverses
+logabsdetjac(ib::Inversed{<: Bijector}, y) = - logabsdetjac(ib.orig, transform(ib, y))
+
+inv(b::Bijector) = Inversed(b)
+inv(ib::Inversed{<:Bijector}) = ib.orig
+
+# AD implementations
+function jacobian(b::ADBijector{<: ForwardDiffAD}, y::Real)
+    ForwardDiff.derivative(z -> transform(b, z), y)
```

Comment:
To be super picky, could you please put a return statement in from of all of these single line function bodies? (See Turing style guide section on method definitions)

Reply:
Sorry! I'l fix all the style issues asap

Reply:
Think I got them all :+1: